examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,274 @@
+"""
+===============================================
+Interpretation of coefficients in linear models
+===============================================
+
+Linear models describe situations in which the target value is expected to be
+a linear combination of the features (see the :ref:`linear_model` User Guide
+section for a description of a set of linear model methods available in
+scikit-learn).
+It is important to emphasize that linear models compute conditional links.
+The interpretation of the coefficients gives the relationship between the
+feature and the target given that other features remain constant.
+
+This example will show some hints in interpreting coefficient in linear models,
+using data from the "Current Population Survey" from 1985.
+"""
+
+print(__doc__)
+
+import numpy as np
+import scipy as sp
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+#############################################################################
+# Determinants of Wages from the 1985 Current Population Survey
+# -------------------------------------------------------------
+#
+# The dataset
+# ...........
+#
+# We fetch the data from `OpenML <http://openml.org/>`_.
+# Note that setting the parameter `as_frame` to `True` will retrieve the data
+# as a pandas dataframe.
+
+from sklearn.datasets import fetch_openml
+
+survey = fetch_openml(data_id=534, as_frame=True)
+
+##############################################################################
+# Then, we identify features (`X`) and targets (`y`): the column 'WAGE' is our
+# target variable (i.e., the variable which we want to predict).
+
+X = survey.data[survey.feature_names]
+print(X.head())
+##############################################################################
+y = survey.target.values.ravel()
+print(survey.target.head())
+
+##############################################################################
+# First, let's get some insights by looking at the marginal links between the
+# different variables. Only numerical variables will be used.
+
+sns.pairplot(survey.frame, diag_kind='kde')
+
+##############################################################################
+# Note that the "WAGE" distribution has a long tail and we could take its log
+# to simplify our problem getting closer to a normal distribution.
+# For all 3 variables, "EDUCATION", "EXPERIENCE", and "AGE", the "WAGE" is
+# increasing when these variables are increasing. Also, the "EXPERIENCE" and
+# "AGE" are correlated.
+#
+# The pipeline
+# ............
+
+survey.data.info()
+
+#############################################################################
+# The dataset is composed by columns with different data types and we need to
+# apply a specific preprocessing for each data types.
+# Our pre-processor will
+#
+# - one-hot encode (i.e., generate a column by category) the categorical
```

Comment:
Done, I'm not sure about how it's formulated though

Reply:
(changes aren't pushed yet so I'm not sure my comment will apply ;)  )

We could say "to avoid categorical features to be treated as ordered values, we need to one-hot-encode them" or something like that.

Reply:
Done